### PR TITLE
RUCIO_Transfers.py: Add RegisterReplicas action

### PR DESCRIPTION
**Pull request by novicecpp**
The second PR of new RUCIO_Transfers.py scripts for `RegisterReplicas` action.
Note that I merged https://github.com/dmwm/CRABServer/pull/7636 and this PR is opened on top of that.

@belforte do you prefer to review it by yourself or do it together ?

### CODE REVIEW

File: src/python/ASO/Rucio/RunTransfer.py [hunk 1]
```diff
@@ -33,14 +34,23 @@ def algorithm(self):
         self.transfer = Transfer()
         self.transfer.readInfo()
         self.rucioClient = self._initRucioClient(self.transfer.username, self.transfer.restProxyFile)
-        #self.crabRESTClient = self._initCrabRESTClient
-        # do dbs dataset
-        #BuildDBSDataset(self.transfer, self.rucioClient).execute()
+        self.transfer.readInfoFromRucio(self.rucioClient)
```
Comment (belforte) [inline src/python/ASO/Rucio/RunTransfer.py:37]: readInfoFromRucio is not a good name. Name should indicate the action, not the means used to perform it. Maybe something like populateLfn2DatasetMap ?

File: src/python/ASO/Rucio/Transfer.py [hunk 2]
```diff
@@ -89,6 +112,15 @@ def readTransferItems(self):
         if len(self.transferItems) == 0:
             raise RucioTransferException(f'{path} does not contain new entry.')
 
+    def buildLFN2IDMap(self):
+        """
+        Create `self.replicaLFN2IDMap` for map from LFN to ID of REST file
+        trasnfer.
```
Comment (belforte) [inline src/python/ASO/Rucio/Transfer.py:118]: add to comment: using info in transferItems

File: src/python/ASO/Rucio/utils.py [hunk 3]
```diff
@@ -14,9 +16,88 @@ def writePath(path):
...
+    :param n: chunk size
+    :type n: int
+    :return: yield the next list chunk
+    :rtype: generator
+    """
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def updateDB(client, api, subresource, fileDoc):
```
Comment (belforte) [inline src/python/ASO/Rucio/utils.py:44]: do we really want a function here ? In the end it is a one line thing

**Comment by belforte** [comment]
Sorry, has a 2h power cut. OK. Let me finish reading and then switch to new version.

**Comment by belforte** [comment]
everything looks good to me. Only the minor comment above.
Close at your convenience

**Comment by novicecpp** [comment]
Closing this in favor of new PR https://github.com/dmwm/CRABServer/pull/7649.

**Pull request closed without merging**